articles/introducing-deity.md

```diff
@@ -0,0 +1,285 @@
+
+Basically, it does this:
+
+```javascript
+deity('number:10-20', function (num) {
+	// This function will be called 100 times
+	// num will equal a random number between 10 and 20 each time
+});
+```
+
+This is useful for testing a range of values against a function you've written,
+instead of just a couple values that you specified when writing the test.
+
+The first argument is what we call a "generator string": it specifies the name
+of the generator that should be used and any arguments to be given to the
+generator—in the case of the number generator, a range to generate the number
+within.
+
+You can specify multiple generator strings as multiple arguments, and the
+callback will be called with multiple arguments.
+
+It then has an optional argument which is an object containing configuration
+for both Deity and any generators. For example, we can specify the number of
+times we want Deity to run the generator.
+
+The final argument is a function, your testing function, which will be called a
+number of times with different generated data each time. You can then run your
+tests on this data.
+
+
+## An example
+
+Let's say we want to test a `reverse()` function that simply takes a string and
+returns it in reverse order. Traditionally with mocha and an assertion library,
+we would write a test for it like this:
+
+```javascript
+describe('reverse()', function () {
+	it('should return reversed string', function () {
+		assert.equal(reverse('abc'), 'cba');
+		assert.equal(reverse('qwerty'), 'ytrewq');
+		assert.equal(reverse('hello'), 'olleh');
+	});
+});
+```
+
+That's all very well, but we're only testing those three specific strings. It
+would be nice if we could generate some strings to test—and with Deity, we can.
+
+Let's rewrite the example to call `reverse()` twice, which should return the
+original string:
+
+```javascript
+describe('reverse()', function () {
+	it('should return reversed string', function () {
+		assert.equal(reverse(reverse('abc')), 'abc');
+		assert.equal(reverse(reverse('qwerty')), 'qwerty');
+		assert.equal(reverse(reverse('hello')), 'hello');
+	});
+});
+```
+
+Now we're testing that the input string and output string are the same instead
+of manually performing the reverse logic when writing our tests. We can use
+Deity to generate the test strings for us:
+
+```javascript
+describe('reverse()', function () {
+	it('should return reversed string', function () {
+		return deity('string:5-10', function (str) {
+			assert.equal(reverse(reverse(str)), str);
+		});
+	});
+});
+```
+
+Our assertion will be called 100 times with a different random string each time
+with a length between 5 and 10 characters. This is far superior to testing the
+same three strings every time!
+
+
+## Generators
+
+Deity has quite a few built in generators which can be used to generate many
+types of data. Here are a few of the most useful:
+
+### String
+
+The string generator generates strings of random length within a specified
```

Comment:
I think you can cut down on the leading paragraphs describing a generator, it can be implied via the title.

E.g.
> ### String
> Generates strings of random length …

> ### String Generator
> Generates strings of random length …